sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/functionResources.scala

```diff
@@ -20,16 +20,24 @@ package org.apache.spark.sql.catalyst.catalog
 import org.apache.spark.sql.AnalysisException
 
 /** An trait that represents the type of a resourced needed by a function. */
-sealed trait FunctionResourceType
+sealed trait FunctionResourceType {
+  def resourceType: String
+}
 
-object JarResource extends FunctionResourceType
+object JarResource extends FunctionResourceType {
+  val resourceType = "jar"
```

Comment:
If we make FunctionResourceType and abstract class
These could be written as `object JarResource extends FunctionResourceType("jar)`
Any thoughts ?